src/envoy/http/service_control/client_cache_test.cc

```diff
@@ -0,0 +1,179 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/envoy/http/service_control/client_cache.h"
+#include "common/common/empty_string.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "test/mocks/common.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/test_common/utility.h"
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace service_control {
+namespace {
+
+using ::espv2::api_proxy::service_control::CheckResponseInfo;
+using ::google::api::envoy::http::service_control::FilterConfig;
+using ::google::api::envoy::http::service_control::Service;
+using ::google::api::servicecontrol::v1::CheckError;
+using ::google::protobuf::util::Status;
+using ::google::protobuf::util::error::Code;
+using ::testing::NiceMock;
+
+class ClientCacheCheckResponseTest : public ::testing::Test {
+ protected:
+  ClientCacheCheckResponseTest() : stats_base_("test", context_.scope_) {
+    token_fn_ = []() { return Envoy::EMPTY_STRING; };
+  }
+
+  void SetUp() override {
+    cache_ = std::make_unique<ClientCache>(
+        service_config_, filter_config_, stats_base_.stats(), cm_, time_source_,
+        dispatcher_, token_fn_, token_fn_);
+  }
+
+  void CheckAndReset(Envoy::Stats::Counter& counter, const int expected_value) {
+    EXPECT_EQ(counter.value(), expected_value);
+    counter.reset();
+  }
+
+  void TearDown() override {
```

Comment:
Seem unnecessary

Reply:
This is needed to ensure we didn't set any other stats other than the ones in the test.